connectors/services/sync.py

```diff
@@ -35,13 +37,13 @@ class SyncService(BaseService):
     def __init__(self, config):
         super().__init__(config)
         self.idling = self.service_config["idling"]
-        self.hb = self.service_config["heartbeat"]
+        self.heartbeat_interval = self.service_config["heartbeat"]
         self.concurrent_syncs = self.service_config.get(
             "max_concurrent_syncs", DEFAULT_MAX_CONCURRENT_SYNCS
         )
         self.bulk_options = self.es_config.get("bulk", {})
         self.source_klass_dict = get_source_klass_dict(config)
-        self.connectors = None
+        self.connector_index = None
         self.sync_job_index = None
         self.syncs = None
```

Comment:
Where is this initialised? I see a `self.syncs.put` call, but no explicit assignment in this file.

Reply:
Indeed looks like a problem!

Reply:
It's assigned outside of changes in this PR:

https://github.com/elastic/connectors-python/blob/b6b23580b202e8ee41f0e0fd5d43f345697d47a0/connectors/services/sync.py#L130